File: easy/1752-check-if-array-is-sorted-and-rotated/solution.py
from typing import List


class Solution:
    def check(self, nums: List[int]) -> bool:
        """
        1752. Check if Array Is Sorted and Rotated
        Time: O(n)
        Space: O(1)
        """
        drops = 0
        n = len(nums)

        for i in range(n):
            if nums[i] > nums[(i + 1) % n]:
                drops += 1

                if drops > 1:
                    return False

        return True

File: easy/1752-check-if-array-is-sorted-and-rotated/solution.java
/**
 * 1752. Check if Array Is Sorted and Rotated
 * Time: O(n)
 * Space: O(1)
 */
class Solution {
    public boolean check(int[] nums) {
        int drops = 0;
        int n = nums.length;

        for (int i = 0; i < n; i++) {
            if (nums[i] > nums[(i + 1) % n]) {
                drops++;

                if (drops > 1) {
                    return false;
                }
            }
        }

        return true;
    }
}
